src/track/beatmap.cpp

```diff
@@ -268,6 +274,93 @@ double BeatMap::findNthBeat(double dSamples, int n) const {
     return -1;
 }
 
+bool BeatMap::findPrevNextBeats(double dSamples,
```

Comment:
It would be nice if we could share the logic between findNthBeat and findPrevNextBeats since for BeatMap it's pretty complicated. What if this was called ```findNthBeats(double dSample, int n1, double* dpBeat1Samples, int n2, double* dpBeat2Samples)``` -- findNthBeat could call that and throw away the 2nd number. Or there could be a helper function that calculates the 3 iterators and then the findPrevNextBeats could do its custom thing. Shrug -- not super urgent so we could wait until after the release to clean up.